src/components/product.js

```diff
@@ -461,7 +477,7 @@ export default class Product extends Component {
    * @return {String}
    */
   get onlineStoreURL() {
-    const identifier = this.handle ? this.handle : this.id;
+    const identifier = this.handle ? this.handle : atob(this.id).split('/').pop();
```

Comment:
you might not, but there are valid use cases to go from the buy button view to the online store view. That's a behaviour that exists today that we can't deprecate.